src/playground-code-editor.ts

```diff
@@ -455,14 +458,25 @@ export class PlaygroundCodeEditor extends LitElement {
           fileContent: this.value,
           tokenUnderCursor,
           cursorIndex: this.cursorIndex,
-          provideCompletions: (completions: EditorCompletion[]) => {
-            this.completions = completions;
-          },
+          provideCompletions: (completions: EditorCompletion[]) =>
+            this._onCompletionsProvided(id, completions),
         },
       })
     );
   }
 
+  private _onCompletionsProvided(id: number, completions: EditorCompletion[]) {
+    // To prevent race conditioning, check that the completions provided
+    // are from the latest completions request.
+    if (id !== this._currentCompletionRequestId) {
+      return;
+    }
+
+    // Reset the counter on a successful completion
+    this._currentCompletionRequestId = 0;
```

Comment:
I don't think this should be reset, since that could cause the same kind of race condition. For example, this sequence:

request A
  current incremented to 1
request B
  current incremented to 2
response B
  B accepted
  current reset to 0
request C
  current incremented to 1
response A
  A accepted

Reply:
Ahh you're right. Somehow it felt like a good idea yesterday. Reverted now